hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java

```diff
@@ -408,21 +437,27 @@ private void assertClusterPrepared(long preparedIndex,
       LambdaTestUtils.await(WAIT_TIMEOUT_MILLIS,
           1000, () -> {
           if (!om.isRunning()) {
+            LOG.info("{} is not yet started.", om.getOMNodeId());
             return false;
           } else {
             boolean preparedAtIndex = false;
             OzoneManagerPrepareState.State state =
                 om.getPrepareState().getState();
 
+            LOG.info("{} has prepare status: {} index: {}.",
+                om.getOMNodeId(), state.getStatus(), state.getIndex());
+
             if (state.getStatus() == PrepareStatus.PREPARE_COMPLETED) {
-              if (state.getIndex() == preparedIndex) {
+              if (state.getIndex() >= expectedPreparedIndex) {
                 preparedAtIndex = true;
               } else {
                 // State will not change if we are prepared at the wrong index.
```

Comment:
IMHO, here we should consider a case: 2 successive prepares(just like the case testPrepareWithTransactions), we have a chance to get PREPARE_COMPLETED of the first round, then its index may be < expectedPreparedIndex of the second round,  because the status of om is async updated(just as we wait for it to update in prepare CLI).
Maybe we could add a retry / sleep / wait for the updated prepare status? @errose28

Reply:
Yes good catch. This is the cause of the intermittent failure in testPrepareWithTransactions. I don't think we can fail fast out of here. We must wait for the transaction index to be at least the expected prepare index to pass. I was able to repro the intermittent failure, and after removing this fail fast case from the code I have not observed the failure.

Reply:
Oh, yes it would be fine if we don't fail fast and only fail the case upon timeout.